sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -75,6 +78,16 @@ private[sql] object JDBCRelation {
     }
     ans.toArray
   }
+
+  def getEffectiveProperties(
+      connectionProperties: Properties,
+      extraOptions: scala.collection.Map[String, String] = Map()): Properties = {
+    val props = new Properties()
+    props.putAll(extraOptions.asJava)
+    // connectionProperties should override settings in extraOptions
+    props.putAll(connectionProperties)
```

Comment:
How does this handle inherited properties? E.g. if `connectionProperties` was defined by inheriting defaults from a parent set of properties, would those inherited defaults be preserved here or would they be lost? This concern existed in the old code as well, but just thought I'd ask since we're using this method in a few places in order to do defensive copying of property objects.